PJ05/src/Exceptions/RequestFailedException.java

```diff
@@ -1,5 +1,18 @@
 package Exceptions;
 
-public class RequestFailedException {
+import Request.Request;
 
+public class RequestFailedException extends Exception {
+    Request request;
+
+    public RequestFailedException() {
+    }
+
+    public RequestFailedException(String message) {
+        super(message);
+    }
+
+    public RequestFailedException(Request request) {
+        this.request = request;
```

Comment:
We might not need this constructor, nor do we need a `Request` property in `RequestFailedException`. Reason: `request.uuid` is already included in `Response`.